internal/app/machined/pkg/system/services/kubelet.go

```diff
@@ -164,7 +168,8 @@ func (k *Kubelet) Runner(r runtime.Runtime) (runner.Runner, error) {
 		{Type: "bind", Destination: "/etc/os-release", Source: "/etc/os-release", Options: []string{"bind", "ro"}},
 		{Type: "bind", Destination: "/etc/cni", Source: "/etc/cni", Options: []string{"rbind", "rshared", "rw"}},
 		{Type: "bind", Destination: "/usr/libexec/kubernetes", Source: "/usr/libexec/kubernetes", Options: []string{"rbind", "rshared", "rw"}},
-		{Type: "bind", Destination: "/var/run", Source: "/run", Options: []string{"rbind", "rshared", "rw"}},
+		{Type: "bind", Destination: "/var/run", Source: "/run/kubelet", Options: []string{"rbind", "rshared", "rw"}},
+		{Type: "bind", Destination: "/var/containerd", Source: "/run/containerd", Options: []string{"rbind", "rshared", "rw"}},
```

Comment:
I _think_ we might be able to get away with keeping the original socket location if we do something like:

```
{Type: "bind", Destination: constants.ContainerdAddress, Source: constants.ContainerdAddress, Options: []string{"bind", "ro"}},
```

Reply:
The error about multiple overlay filesystems might be due to the fact that we are trying to mount a dir, vs a file.

Reply:
thanks, I'll try that!

Reply:
looks like same error when both mounts end up in the same tree:

```
 service[kubelet](Waiting): Runner Containerd(kubelet) exited without error, going to restart it
[  141.519972] [talos] service[kubelet](Waiting): Error running Containerd(kubelet), going to restart forever: failed to create task: "kubelet": OCI runtime create failed: container_linux.go:349: starting container process caused "process_linux.go:449: container init caused \"rootfs_linux.go:58: mounting \\\"/run/containerd\\\" to rootfs \\\"/run/containerd/io.containerd.runtime.v2.task/k8s.io/kubelet/rootfs\\\" at \\\"/run/containerd/io.containerd.runtime.v2.task/k8s.io/kubelet/rootfs/run/containerd\\\" caused \\\"no such file or directory\\\"\"": unknown
[  146.547418] overlayfs: upperdir is in-use as upperdir/workdir of another mount, accessing files from both mounts will result in undefined behavior.
[  146.549122] overlayfs: workdir is in-use as upperdir/workdir of another mount, accessing files from both mounts will result in undefined behavior.
```

Reply:
trying to debug that, I wonder if that is some error with our containerd runner

Reply:
Strange. Ok so I was wrong then. I guess overlay is used regardless. I'm trying to think of a way we can secure this but keep the containerd socket the default path. Will keep thinking.

Reply:
figured it out, `rshared` was a problem, as this was creating a loop

Reply:
with `rshared`, on the host side we had `/run/containerd` mounted under `/run/kubelet`, so on kubelet restart two mounts were layered one on top of another (kind of duplicate mount?)